zephyr: namespace the generated version.h

ycsin opened the pull request:
Zephyr's build time generated `version.h` is now in the `zephyr` to provide proper namespace, update the includes of this module accordingly.

See https://github.com/zephyrproject-rtos/zephyr/pull/63973

See also https://github.com/thesofproject/sof/pull/8502#issuecomment-1821242038:

> @ycsin So can you get the Zephyr thing merged by reopening [zephyrproject-rtos#31](https://github.com/zephyrproject-rtos/sof/pull/31) and use that in Zephyr west.yml, and when we update SOF to new Zephyr, we'll merged this PR as part of the west.yml update. I think that's what @marc-hb wanted , and this, too, is ok for me. If Zephyr maintainers ask, you can point to this comment.

nashif committed:
app: zephyr still requires cmake version 3.20.0

Set min. cmake version required to what zephyr uses.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

dcpleung committed:
zephyr: use k_smp_cpu_start/_resume for secondary core power up

This changes the seconday core power up routine to use the newly
introduced k_smp_cpu_start() and k_smp_cpu_resume(). This removes
the need to mirror part of the SMP start up code from Zephyr, and
no longer need to call into Zephyr private kernel code.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
zephyr: cpu: use arch_cpu_start instead of arch_start_cpu

This uses arch_cpu_start() instead of arch_start_cpu() as it has
been renamed to align with the CPU namespace.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

nashif committed:
zephyr: use arch_xtensa_uncached_ptr/arch_xtensa_cached_ptr

Use direct call to function instead of going over a redefine.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

ycsin committed:
build: namespace the generated headers with `zephyr/``
Zephyr's build time generated headers are now in the
`zephyr` to provide proper namespace, update the path
accordingly.

See ZephyrProject upstream PR at:
zephyrproject-rtos/zephyr#63973

Signed-off-by: Yong Cong Sin <ycsin@meta.com>

kv2019i reviewed (approved):
Thanks @ycsin !

ycsin commented:
@kv2019i if I'm not wrong, is this PR supposed to be merged before #63973, so that #63973 can update the revision to point at this commit?

marc-hb reviewed (approved):
> @kv2019i if I'm not wrong, is this PR supposed to be merged before #63973, so that #63973 can update the revision to point at this commit?

100% yes.

Also, merging this PR #34 will have no effect until some west.yml points at it.

kv2019i commented:
@nashif This is ready to go to enable merging of https://github.com/zephyrproject-rtos/zephyr/pull/63973 . We'll cherry-pick this patch to upstream SOF when we update Zephyr after 63973 is merged.

nashif committed:
app: fix filter in sample.yaml

nxp_adsp_imx8ulp needs to be in allowed platforms.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
Revert "zephyr: use arch_xtensa_uncached_ptr/arch_xtensa_cached_ptr"

This reverts commit b4214e582080103532c89dd68c35a65528c7f139.

nashif committed:
zephyr: use system cache APIs instead of custom soc calls

Use system cache API instead of soc specific calls.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

marc-hb commented:
The target branch has been force-pushed, so you must rebase.

This is something GitHub should _really_ send notifications about. But Github doesn't care much about force-pushes in general (https://github.com/zephyrproject-rtos/zephyr/issues/53566), so it doesn't.

golowanow committed:
zephyr: app: scripts: intel_adsp: change board names to HWMv2

Change `intel_adsp` board names to HWMv2 scheme:

  `intel_adsp_cavs25`     --> `intel_adsp/cavs25`
  `intel_adsp_ace15_mtpm` --> `intel_adsp/ace15_mtpm`
  `intel_adsp_ace20_lnl`  --> `intel_adsp/ace20_lnl`

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
zephyr: intel_adsp: Change ACE SoC name to HWMv2

SoC series ACE changes its name to align with HWMv2 new naming
conventions: SOC_SERIES_INTEL_ACE --> SOC_SERIES_INTEL_ADSP_ACE

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

iuliana-prodan committed:
zephyr: sof: update board name for HWMv2

Update NXP board names and configs with
the ones for HWMv2.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

LaurentiuM1234 committed:
nxp: imx8ulp: change SOC name to MIMX8UD7

Zephyr PR#70219 changes i.MX8ULP's SOC name to MIMX8UD7.
As such, we need to update the board configuration file
and the SOC name inside `xtensa-build-zephyr.py` to use
the new SOC name.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

LaurentiuM1234 committed:
zephyr: CMakeLists.txt: use new `CONFIG_SOC_` for 8ULP

Zephyr PR#70219 changes the SOC name of i.MX8ULP from MIMX8ULP
to MIMX8UD7. This means that all `CONFIG_SOC_*` configurations
will be changed from the `MIMX8ULP`-based naming to the
`MIMX8UD7`-based naming. As such, this commit updates the name
of the configuration used by the `zephyr/CMakeLists.txt` file.

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

iuliana-prodan committed:
zephyr: sof: update configs for NXP ADSP

CONFIG_SOC_<name> has changed in zephyr in order to
match soc name.
Threfore, update configs with new values.

Signed-off-by: Iuliana Prodan <iuliana.prodan@nxp.com>

marc-hb reviewed (approved):
Approved as long as it builds successfully.

kv2019i commented:
I needed to do some adaptation already now to SOF side build scripts as the header location has changed for e.g. autoconf.h -> https://github.com/thesofproject/sof/pull/9179
